pat/pat.go

```diff
@@ -283,6 +284,31 @@ func (p *Pattern) String() string {
 }
 
 /*
+URLPath replaces the named matches in the Pattern with the values provided
+and returns the resulting string. It can be used to generate a URL
+that would match the given pattern with the provided variables. You must
+provide a mapping for all named matches defined in the Pattern.
+*/
+func (p *Pattern) URLPath(vars map[pattern.Variable]string) (string, error) {
+	var missing []string
+
+	path := patternRe.ReplaceAllStringFunc(p.raw, func(s string) string {
```

Comment:
I'd probably prefer looping through `pats` and `literals` here: it's a little bit ugly, but I suspect it'll be more efficient, and the error handling becomes a bit nicer since we don't need to deal with escaping the callback.